src/mapping/impl/BasisFunctions.hpp

```diff
@@ -3,30 +3,40 @@
 #if defined(__NVCC__)
 
 #include <cuda_runtime.h>
+#include <ginkgo/extensions/kokkos.hpp>
+#include <ginkgo/ginkgo.hpp>
+
 #define BOOST_PP_VARIADICS 1
 #define PRECICE_HOST_DEVICE __host__ __device__
 #define PRECICE_MEMORY_SPACE __device__
-#define FMA fma
+#define FMA Kokkos::fma
+#define LOG Kokkos::log
 
 #elif defined(__HIPCC__)
 
-// #define __HIP_PLATFORM_AMD__
+#include <ginkgo/extensions/kokkos.hpp>
+#include <ginkgo/ginkgo.hpp>
 #include <hip/hip_runtime.h>
+
 #define PRECICE_HOST_DEVICE __host__ __device__
 #define PRECICE_MEMORY_SPACE __device__
-#define FMA fma
+#define FMA Kokkos::fma
+#define LOG Kokkos::log
 
 #else
 
 #define PRECICE_HOST_DEVICE
 #define PRECICE_MEMORY_SPACE
 #define FMA std::fma
+#define LOG std::log
 
 #endif
 
-PRECICE_MEMORY_SPACE const double NUMERICAL_ZERO_DIFFERENCE = 1.0e-14;
+#define NUMERICAL_ZERO_DIFFERENCE_DEVICE 1.0e-14
```

Comment:
`constexpr double` is the modern way of doing this.

Reply:
Yes, I did it this way because the reference executor with Kokkos didn't want to accept the `constexpr double` and whenever used, the function just returned `nan`. Using a `#define` was the only solution I could find at this point.